activerecord/lib/arel/visitors/postgresql.rb

```diff
@@ -78,6 +78,25 @@ def visit_Arel_Nodes_IsDistinctFrom(o, collector)
           visit o.right, collector
         end
 
+        # Postgres-specific implementation that uses `col = any('{1,2}')` instead of `col IN (1,2)`
+        # to avoid pg_stat_statements churn
+        def visit_Arel_Nodes_HomogeneousIn(o, collector)
+          visit o.left, collector
+
+          if o.type == :in
+            collector << " = any("
+          else
+            collector << " != all("
+          end
+
+          type_caster = ActiveRecord::ConnectionAdapters::PostgreSQL::OID::Array.new(o.attribute.type_caster, ",")
+          values = [type_caster.serialize(o.values)]
```

Comment:
```suggestion
          values = [type_caster.serialize(o.casted_values)]
```

This resolves https://github.com/rails/rails/pull/49388#issuecomment-1736429993